great_expectations/profile/json_schema_profiler.py

```diff
@@ -0,0 +1,253 @@
+import abc
+from enum import Enum
+from typing import Any, Dict, Optional, Union
+
+from great_expectations.core import (
+    ExpectationConfiguration,
+    ExpectationKwargs,
+    ExpectationSuite,
+)
+
+
+class JsonSchemaTypes(Enum):
+    STRING = "string"
+    INTEGER = "integer"
+    NUMBER = "number"
+    ARRAY = "array"
+    NULL = "null"
+    BOOLEAN = "boolean"
+    OBJECT = "object"
+    ENUM = "enum"
+
+
+class NonDataProfiler(object, metaclass=abc.ABCMeta):
+    """
+    A base profiler that creates suites from non-data sources of truth.
+
+    Implementation notes:
+    - . _profile() is required
+    - .validate() is optional if additional validation is desired. Be sure to
+        call super().validate() before the child class custom logic.
+    """
+
+    def __init__(self, configuration: dict = None):
+        self.configuration = configuration
+
+    def validate(self, schema: dict) -> bool:
+        return isinstance(schema, dict)
+
+    def profile(self, schema: dict, suite_name: str) -> ExpectationSuite:
```

Comment:
let's make suite_name optional and more closely mirror the profile API of DataAssetProfiler (e.g. accepting the profiler configuration in the base class).

Reply:
Dumb question what would we name the suite?

Reply:
Answer: In the base API it's optional and in this specific profiler it's required.